### PR TITLE
dedup: TransactionList and BlockResponse

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Ref #440 and #441

## Solution

Removed `TransactionList` from alloy_network

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/network/src/lib.rs [hunk 1]
```diff
@@ -33,24 +34,12 @@ pub use any::AnyNetwork;
...
-    /// Special case for uncle response
-    Uncled,
-}
-
 /// A block response
 #[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
 pub struct BlockResponse<N: Network> {
     #[serde(flatten)]
     header: N::HeaderResponse,
-    transactions: TransactionList<N::TransactionResponse>,
```
Comment (prestwich) [inline crates/network/src/lib.rs:53]: we should probably do the following while we're here:
- delete this block response as unused and close #441 
- make `TransactionList` generic over `T` if it's not already

**Review by prestwich — CHANGES_REQUESTED** [review]

File: crates/network/src/lib.rs (hunk 1 above)
Comment (yash-atreya) [reply crates/network/src/lib.rs:53]: `TransactionList` is already generic over T, I was removing it in favour of `BlockTransactions` (which does have a generic) from `rpc-types`. 

Also if we're removing `BlockResponse` we can also remove `TransactionList` as it is the only place it is used

Comment (yash-atreya) [reply crates/network/src/lib.rs:53]: I'll leave `TransactionsList` as is but would like to change the `Hydrated` enum to `Full` to be consistent with `BlockTransactions`

Comment (prestwich) [reply crates/network/src/lib.rs:53]: i feel like we're having a miscommunication here so i'm gonna step back

- the 2 types in alloy-network lib.rs should be deleted as they're duplicative
- the types in alloy-rpc-types should have generics added with defaults

Comment (yash-atreya) [reply crates/network/src/lib.rs:53]: Sounds great! That's ideal.

**Review by prestwich — APPROVED** [review]

**Pull request merged**